src/btree/bt_curprev.c

```diff
@@ -86,9 +86,9 @@ __cursor_skip_prev(WT_CURSOR_BTREE *cbt)
              * Compiler may replace the usage of the variable with another read in the following
              * code.
              *
-             * Place a read barrier to avoid this issue.
+             * Place a compiler barrier to avoid this issue.
              */
-            WT_ORDERED_READ(ins, cbt->ins_head->head[i]);
+            WT_READ_ONCE(ins, cbt->ins_head->head[i]);
```

Comment:
I'm not sure the comment is enough to ensure this needs to be compiler only. I imagine this is similar to other cases, where a CPU speculatively pre-fetching `cbt->ins_head->head[i]` would also lead to surprises.
I'm not inclined to move to this WT_READ_ONCE notion at all at the moment, but rather stick with the code we have. I could be convinced though, if you are confident in the change.

Reply:
This is OK because cbt->ins_head is thread local. It's not a shared memory.

Reply:
add this to the comment.